walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,65 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Base cases
+    if (request.getAttendees().isEmpty()) { return new ArrayList<>(Arrays.asList(TimeRange.WHOLE_DAY)); }
+    if (request.getDuration() >= TimeRange.WHOLE_DAY.duration()) { return new ArrayList<>(); }
+    
+    // Preprocess event input
+    List<Event> pertinentEvents = new ArrayList<>(filterEvents(new ArrayList<>(events), request.getAttendees()));
+    Collections.sort(pertinentEvents, (Event e1, Event e2) -> TimeRange.ORDER_BY_END.compare(e1.getWhen(), e2.getWhen()));
```

Comment:
I think you can also do it like: Collections.sort(pertinentEvents, TimeRange.ORDER_BY_END);
for future reference (don't need to write your own compare method

Reply:
Can you still do this with List<Event>? I was having trouble figuring out how to make it simpler

Reply:
Actually I think you're right -- my bad I thought it was List<TimeRange> not Event